subprojects/plugin-development/src/integTest/groovy/org/gradle/plugin/devel/impldeps/GradleImplDepsCompatibilityIntegrationTest.groovy

```diff
@@ -128,16 +134,15 @@ class GradleImplDepsCompatibilityIntegrationTest extends BaseGradleImplDepsInteg
         file('src/test/groovy/BuildLogicFunctionalTest.groovy') << """
             import org.gradle.testkit.runner.GradleRunner
             import static org.gradle.testkit.runner.TaskOutcome.*
-            import org.junit.Rule
-            import org.junit.rules.TemporaryFolder
             import spock.lang.Specification
+            import spock.lang.TempDir
 
             class BuildLogicFunctionalTest extends Specification {
-                @Rule TemporaryFolder testProjectDir = new TemporaryFolder()
+                @TempDir File testProjectDir
```

Comment:
Don't we have our own `TestFile` as well?  I agree with removing JUnit's `TemporaryFolder`, but not sure which is a better replacement.

Reply:
Well, with 2.2 Spock supports custom classes, but there is an issue if they extend from `Path`/`File` which was fixed in https://github.com/spockframework/spock/pull/1519 and will be released with 2.3

Reply:
I went with the simplest option. Given that this is a nested test, I don't think using `TestFile` would be appropriate here.